File: src/com/zoarial/iot/models/IoTPacketSectionList.java
package com.zoarial.iot.models;

import java.util.ArrayList;
import java.util.List;
import java.util.UUID;

public class IoTPacketSectionList extends ArrayList<IoTPacketSection> {

    public IoTPacketSectionList() {
        super();
    }

    public IoTPacketSectionList(int initialLength) {
        super(initialLength);
    }

    public void add(String str) {
        super.add(new IoTPacketSection(str));
    }

    public void add(int i) {
        super.add(new IoTPacketSection(i));
    }

    public void add(byte b) {
        super.add(new IoTPacketSection(b));
    }

    public void add(long l) {
        super.add(new IoTPacketSection(l));
    }

    public void add(UUID uuid) {
        super.add(new IoTPacketSection(uuid));
    }

    public byte[] getNetworkResponse() {
        byte[] fullBytesArr;
        int len = 0;

        for(IoTPacketSection section : this) {
            len += section.getByteList().length;
        }

        fullBytesArr = new byte[len];
        len = 0;
        for(IoTPacketSection section : this) {
            System.arraycopy(section.getByteList(), 0, fullBytesArr, len, section.getByteList().length);
            len += section.getByteList().length;
        }

        return fullBytesArr;
    }
}

File: src/main/java/com/zoarial/iot/network/IoTPacketSectionList.kt
package com.zoarial.iot.network

import java.util.*

class IoTPacketSectionList : ArrayList<IoTPacketSection> {
    constructor() : super()
    constructor(initialLength: Int) : super(initialLength) {}

    fun add(str: String) {
        super.add(IoTPacketSection(str))
    }

    fun add(b: Boolean) {
        super.add(IoTPacketSection(b))
    }

    fun add(i: Int) {
        super.add(IoTPacketSection(i))
    }

    fun add(b: Byte) {
        super.add(IoTPacketSection(b))
    }

    fun add(l: Long) {
        super.add(IoTPacketSection(l))
    }

    fun add(uuid: UUID) {
        super.add(IoTPacketSection(uuid))
    }

    val networkResponse: ByteArray
        get() {
            val fullBytesArr: ByteArray
            var len = 0
            for (section in this) {
                len += section.byteList.size
            }
            fullBytesArr = ByteArray(len)
            len = 0
            for (section in this) {
                System.arraycopy(section.byteList, 0, fullBytesArr, len, section.byteList.size)
                len += section.byteList.size
            }
            return fullBytesArr
        }
}